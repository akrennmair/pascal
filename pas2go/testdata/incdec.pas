program test;

var i : integer;

begin
	i := 0;
	inc(i);
	if i <> 1 then
		writeln('Error: i should be 1, is actually ', i);
	inc(i, 23);
	if i <> 24 then
		writeln('Error: i should be 24, is actually ', i);
	dec(i);
	if i <> 23 then
		writeln('Error: i should be 23, is actually ', i);
	dec(i, 13);
	if i <> 10 then
		writeln('Error: i should be 23, is actually ', i)
end.
