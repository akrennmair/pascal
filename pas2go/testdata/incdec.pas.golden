package main

import (
	"github.com/akrennmair/pascal/pas2go/system"
)

var _ = system.Write

// program test
func main() {
	var (
		i int
	)
	_ = i

	i = 0
	i++
	if i != 1 {
		system.Writeln("Error: i should be 1, is actually ", i)
	}
	i += 23
	if i != 24 {
		system.Writeln("Error: i should be 24, is actually ", i)
	}
	i--
	if i != 23 {
		system.Writeln("Error: i should be 23, is actually ", i)
	}
	i -= 13
	if i != 10 {
		system.Writeln("Error: i should be 23, is actually ", i)
	}
}
